test/e2e/konnect_test.go

```diff
@@ -190,28 +192,65 @@ func createTestRuntimeGroup(ctx context.Context, t *testing.T) string {
 			req.Header.Set("Authorization", "Bearer "+konnectAccessToken)
 			return nil
 		}),
-		rg.WithHTTPClient(helpers.RetryableHTTPClient(helpers.DefaultHTTPClient())),
 	)
 	require.NoError(t, err)
 
-	createRgResp, err := rgClient.CreateRuntimeGroupWithResponse(ctx, rg.CreateRuntimeGroupRequest{
-		Description: lo.ToPtr("This is a description"),
-		Labels:      &rg.Labels{"created_in_tests": "true"},
-		Name:        uuid.NewString(),
-		ClusterType: rg.ClusterTypeKubernetesIngressController,
-	})
-	require.NoError(t, err, "failed to create runtime group")
-	require.Equalf(t, http.StatusCreated, createRgResp.StatusCode(), "failed creating RG: %s", string(createRgResp.Body))
-	require.NotNil(t, createRgResp.JSON201)
-	require.NotNil(t, createRgResp.JSON201.Id)
-	id := *createRgResp.JSON201.Id
+	rgName := uuid.NewString()
+	var rgID uuid.UUID
+
+	createRgErr := retry.Do(func() error {
+		listRgResp, err := rgClient.ListRuntimeGroupsWithResponse(ctx, &rg.ListRuntimeGroupsParams{
+			FilterNameEq: lo.ToPtr(rgName),
+		})
+		if err != nil {
+			return fmt.Errorf("failed to list runtime groups with name %s: %w", rgName, err)
+		}
+		if listRgResp.StatusCode() != http.StatusOK {
+			return fmt.Errorf("failed to list RGs: code %d, message %s", listRgResp.StatusCode(), string(listRgResp.Body))
+		}
+
+		if listRgResp.JSON200 != nil && listRgResp.JSON200.Data != nil {
+			// REVIEW: what to do if the runtime group already exists?
+			// considering we are using a random UUID as runtime group name,
+			// the probability of conflict with other existing RGs is very small,
+			// so we use the RG here.
```

Comment:
I'd say not reuse it and fall back regenerate the name and try again with that. Someone using that already existing RG might be surprised that it got deleted after our tests complete.

BTW: is there a way to add labels or tags to those RGsS

Reply:
Updated. Now it creates an RG with new name on failures, and delete all RGs created in the function in Cleanup.